src/main/java/Task/Deadline.java

```diff
@@ -0,0 +1,88 @@
+package Task;
+
+import CommandLine.Line;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    private LocalDateTime deadline;
+    private static Line line = new Line();
+
+    private Deadline(String name, TaskType taskType, LocalDateTime deadline) {
+        super(name, taskType);
+        this.deadline = deadline;
+    }
+
+    /**
+     * Returns the task type as a string
+     * @return task type as a string
+     */
+    @Override
+    public String getTaskTypeAsString(){
+            return "D";
+    }
+
+    /**
+     * Creates a deadline with given string
+     * @param name the string containing information about the task type
+     * @param taskType the type of task
+     * @throws TaskCreationException if error occurs while creating task
+     * @return Deadline
+     */
+    public static Deadline of(String name, TaskType taskType) throws TaskCreationException {
+        try {
+            String[] parts = name.split("/by", 2);
+            String taskName = parts[0].trim();
+            String taskDeadline = parts[1].trim();
+
+            LocalDateTime deadline = formatter(taskDeadline);
+            return new Deadline(taskName, taskType, deadline);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            line.drawLine();
+            System.out.println("      Error ");
+            line.drawLine();
+            throw new TaskCreationException();
+        } catch (DateTimeParseException e) {
+            line.drawLine();
+            System.out.println("      Invalid deadline format. Expected format: 'task description /by date/time' ");
+            line.drawLine();
+            throw new TaskCreationException();
+        }
+    }
+
+    /**
+     * Returns the deadline formatted as a string
+     * @return the deadline as a string
+     */
+    public String getBy() {
```

Comment:
Perhaps a more reader-friendly method name could be used here? For example, `getDeadline` instead of `getBy`